python-sdk/src/astro/databases/base.py

```diff
@@ -305,50 +360,65 @@ def load_file_to_table(
+            columns_names_capitalization=columns_names_capitalization,
+            if_exists=if_exists,
+        )
+
+        if use_native_support and self.is_native_load_file_available(
+            source_file=input_file, target_table=output_table
+        ):
+
+            self.load_file_to_table_natively_with_fallback(
+                source_file=input_file,
+                target_table=output_table,
+                if_exists="append",
+                normalize_config=normalize_config,
+                native_support_kwargs=native_support_kwargs,
+                enable_native_fallback=enable_native_fallback,
+                chunk_size=chunk_size,
+            )
+        else:
+            self.load_file_to_table_using_pandas(
+                input_file=input_file,
+                output_table=output_table,
+                normalize_config=normalize_config,
+                if_exists="append",
+                chunk_size=chunk_size,
+            )
+
+    def load_file_to_table_using_pandas(
+        self,
+        input_file: File,
+        output_table: Table,
+        normalize_config: dict | None = None,
+        if_exists: LoadExistStrategy = "replace",
+        chunk_size: int = DEFAULT_CHUNK_SIZE,
+    ):
         input_files = resolve_file_path_pattern(
             input_file.path,
             input_file.conn_id,
             normalize_config=normalize_config,
             filetype=input_file.type.name,
         )
-        self.create_schema_if_needed(output_table.metadata.schema)
-        if if_exists == "replace" or not self.table_exists(output_table):
-            self.drop_table(output_table)
-            self.create_table(
-                output_table,
-                input_files[0],
-                columns_names_capitalization=columns_names_capitalization,
-            )
-            if_exists = "append"
 
-        # TODO: many native transfers support the input_file.path - it may be better
-        # to use the native support to loading multiple files as opposed to iterating
-        # here
         for file in input_files:
-            if use_native_support and self.is_native_load_file_available(
-                source_file=file, target_table=output_table
-            ):
-                self.load_file_to_table_natively_with_fallback(
-                    source_file=file,
-                    target_table=output_table,
-                    if_exists=if_exists,
-                    native_support_kwargs=native_support_kwargs,
-                    enable_native_fallback=enable_native_fallback,
-                    chunk_size=chunk_size,
-                )
-            else:
-                self.load_pandas_dataframe_to_table(
-                    file.export_to_dataframe(),
-                    output_table,
-                    chunk_size=chunk_size,
-                    if_exists="append",  # We've already created a new table in this case
-                )
+            self.load_pandas_dataframe_to_table(
```

Comment:
@sunank200 What will happen when don't get results after pattern resolution? Do we raise an error or not? If not, I think we should. WDYT?